<launch>
  <!-- launch the camera -->
  <node ns="camera" pkg="uvc_camera" type="camera_node" name="uvc_camera" output="screen">
    <param name="width" type="int" value="1280" />
    <param name="height" type="int" value="720" />
    <!--param name="width" type="int" value="640" />
    <param name="height" type="int" value="480" /-->
    <!--param name="width" type="int" value="1920" />
    <param name="height" type="int" value="1080" /-->
    <param name="fps" type="int" value="24" />
    <!--param name="frame" type="string" value="webcam" /-->
    <param name="device" type="string" value="/dev/video0" />
    <param name="camera_info_url" type="string" value="file://$(find tuROSOpenCV)/logiHD.yaml" />
    <param name="saturation" type="int" value="128" />
    <param name="sharpness" type="int" value="128" />
    <param name="blacklightCompen" type="int" value="1" />
	<param name="brightness" type="int" value="128"/>
	<param name="contrast" type="int" value="128"/>
	<param name="gain" type="int" value="32"/>
	<param name="autoExposure" type="int" value="3"/>
	<!-- autoPriority is the key param controlling the quality of the video grabbed from the camera, pls make sure the value of it is set to int(3) -->
	<param name="autoPriority" type="int" value="1"/>
    <!--param name="mode" type="string" value="MJPG" /--> <!-- mode:MJPG RGB YUYV-->
    <param name="mode" type="string" value="RGB" /> <!-- mode:MJPG RGB YUYV-->
    <!--param name="mode" type="string" value="YUYV" /--> <!-- mode:MJPG RGB YUYV-->
  </node>

  <!-- view the grabbed image-->
  <node name="image_view_raw" pkg="image_view" type="image_view">
		  <param name="autosize" value="true"/>
		  <remap from="image" to="/camera/image_raw"/>
  </node>


  <!-- launch the endeffector_tracking pkg-->
  <node name="endeffector_tracking" pkg="endeffector_tracking" type="track_node">
		  <param name="camera_topic" type="string" value="camera/image_raw"/>
		  <param name="config_file" type="string" value="file://$(find endeffector_tracking)/config.xml"/>
		  <!--param name="model_name" type="string" value="$(find endeffector_tracking)/config.wrl"/-->
		  <param name="model_name" type="string" value="$(find endeffector_tracking)/config.cao"/>
		  <param name="init_file" type="string" value="$(find endeffector_tracking)/config.init"/>
  </node>

  <!-- launch the hough viewer-->
  <node name="hough_view" pkg="image_view" type="image_view">
		  <param name="autosize" value="true"/>
		  <remap from="image" to="/hough"/>
  </node>
</launch>

